rerun_cpp/src/rerun/datatypes/quaternion_ext.cpp

```diff
@@ -18,6 +18,11 @@ namespace rerun {
             /// Construct Quaternion from x/y/z/w values.
             Quaternion(float x, float y, float z, float w) : xyzw{x, y, z, w} {}
 
+            /// Construct Quaternion from x/y/z/w float pointer.
+            ///
+            /// Attention: The pointer must point to at least least 4 floats long.
+            Quaternion(const float* ptr) : xyzw{ptr[0], ptr[1], ptr[2], ptr[3]} {}
```

Comment:
We can have a more useful name than "pointer" (we already know it is a pointer). `xyzw_` helps the user see the order of the elements.

```suggestion
            Quaternion(const float* xyzw_) : xyzw{xyzw_[0], xyzw_[1], xyzw_[2], xyzw_[3]} {}
```

Reply:
Actually, I think we should scrap both these constructors and have explicit `from_xyzw` static constructor instead, because `w` last is _not_ universal. We already do this in Python.

It can still be overloaded.